digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java

```diff
@@ -905,16 +906,50 @@ public void addResumingTasks(long attemptId, List<ResumingTask> tasks)
         }
 
         @DigdagTimed(value = "dtcst_", category = "db", appendMethodName = true)
+        /**
+         * Filter the child task if it is generated sub task.
+         * Check the child task full name has '^sub' position later the parent full name.
+         * @param parentFullName
+         * @return
+         */
+        @VisibleForTesting
+        public Predicate<StoredTask> funcFilterGeneratedSubtasks(Optional<String>parentFullName)
+        {
+            Predicate<StoredTask> filterGeneratedSubtasks = (s) -> {
+                if (!parentFullName.isPresent()) {
+                    return true;
+                }
+                String pfname = parentFullName.get();
+                String chfname = s.getFullName();
+                int pfnameLen = pfname.length();
+                int chfnameLen = chfname.length();
+
+                if (chfname.startsWith(pfname)
+                        && chfname.substring(pfnameLen).contains("^sub")) {
+                    // && chfnameLen > pfnameLen + "^sub".length()) {
+                    logger.debug("Exclude generated task:{} from retry target", chfname);
```

Comment:
These code may be better to remove.

Reply:
I deleted here. https://github.com/treasure-data/digdag/pull/1345/commits/1071cbf07425d0ddd1049138d0e696fcac9705c9